src/main/java/hudson/plugins/release/pipeline/ReleaseStep.java

```diff
@@ -0,0 +1,160 @@
+    @Nonnull
+    private List<ParameterValue> parameters;
+
+    @DataBoundConstructor
+    public ReleaseStep(String job) {
+        this.job = job;
+        parameters = new ArrayList<>();
+    }
+
+    public String getJob() {
+        return job;
+    }
+
+    public void setJob(String job) {
+        this.job = job;
+    }
+
+    public List<ParameterValue> getParameters() {
+        return parameters;
+    }
+
+    @DataBoundSetter
+    public void setParameters(List<ParameterValue> parameters) {
+        if (parameters == null) {
+            this.parameters = new ArrayList<>();
+        } else {
+            this.parameters = parameters;
+        }
+    }
+
+    @Override
+    public StepExecution start(StepContext stepContext) throws Exception {
+        return new ReleaseStepExecution(stepContext, this);
+    }
+
+    @Extension
+    public static class DescriptorImpl extends StepDescriptor {
+
+        @Override
+        public String getFunctionName() {
+            return "release";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Trigger release for the job";
+        }
+
+        @Override
+        public Set<Class<?>> getRequiredContext() {
+            return ImmutableSet.of(Run.class, TaskListener.class);
+        }
+
+        @Override
+        public Step newInstance(StaplerRequest req, JSONObject formData) throws FormException {
+            ReleaseStep step = (ReleaseStep) super.newInstance(req, formData);
+            // Cf. ParametersDefinitionProperty._doBuild:
+            Object parameter = formData.get("parameter");
+            JSONArray params = parameter != null ? JSONArray.fromObject(parameter) : null;
+            if (params != null) {
+                Jenkins jenkins = Jenkins.getInstance();
+                Job<?,?> context = StaplerReferer.findItemFromRequest(Job.class);
+                Job<?,?> job = jenkins != null ? jenkins.getItem(step.getJob(), context, Job.class) : null;
+
+                BuildableItem project = Jenkins.getActiveInstance()
+                                               .getItem(step.getJob(), context, BuildableItem.class);
+
+                if (project instanceof BuildableItemWithBuildWrappers) {
+                    ReleaseWrapper wrapper = ((BuildableItemWithBuildWrappers) project).getBuildWrappersList()
+                                                                       .get(ReleaseWrapper.class);
+                    if (wrapper == null) {
+                        throw new IllegalArgumentException("Job doesn't have release plugin configuration");
+                    }
+                    List<ParameterDefinition> parameterDefinitions = wrapper.getParameterDefinitions();
+                    if (parameterDefinitions != null) {
+                        List<ParameterValue> values = new ArrayList<>();
+                        for (Object o : params) {
+                            JSONObject jo = (JSONObject) o;
```

Comment:
This is an unsafe cast 🐛  Will throw exception in the case o null field or array